sqlmesh/core/model/meta.py

```diff
@@ -391,3 +422,18 @@ def when_matched(self) -> t.Optional[exp.When]:
         if isinstance(self.kind, IncrementalByUniqueKeyKind):
             return self.kind.when_matched
         return None
+
+    @property
+    def catalog(self) -> t.Optional[str]:
+        """Returns the catalog of a model."""
+        return self.fqt.catalog
+
+    @property
+    def fqt(self) -> exp.Table:
+        return exp.to_table(self.fqn)
+
+    @property
+    def fqn(self) -> str:
+        return normalize_model_name(
+            self.name, default_catalog=self.default_catalog, dialect=self.dialect
```

Comment:
this doesn't look right, self.name is already normalized, this is performing a double normalization

Reply:
also let's cache this as it seems expensive